### ISSUE TITLE
Azure TrafficManager authentication exception

### ISSUE THREAD

**Issue by viranch** [octodns/octodns-azure#23]
Reported here https://github.com/octodns/octodns-azure/pull/19#issuecomment-1033120128

```
(env) coho:octodns-azure ross$ PYTHONPATH=. octodns-sync --config-file=config/dev.yaml --doit
...
  File "/Users/ross/octodns/octodns-azure/octodns_azure/__init__.py", line 607, in _populate_traffic_managers
    list_profiles = self._tm_client.profiles.list_by_resource_group
  File "/Users/ross/octodns/octodns-azure/octodns_azure/__init__.py", line 565, in _tm_client
    self.__tm_client = TrafficManagerManagementClient(
TypeError: __init__() got an unexpected keyword argument 'credential'
```

```
(env) coho:octodns-azure ross$ pip freeze | grep azure
azure-common==1.1.27
azure-core==1.21.1
azure-identity==1.7.1
azure-mgmt-core==1.3.0
azure-mgmt-dns==8.0.0
azure-mgmt-trafficmanager==0.51.0
msrestazure==0.6.4
```

**Comment by viranch** [comment]
Looks like setup.py is allowing the latest version of `azure-mgmt-trafficmanager` to be installed, and the latest (as of today) happens to be a new major version.

https://github.com/octodns/octodns-azure/blob/252a2f039320877f9df4b504f51a08af42e6c312/setup.py#L41

I didn't realize that this caused my testing setup during #19 to install v1.0.0 🤦 

```
$ pip freeze | grep traffic
azure-mgmt-trafficmanager==1.0.0
```

**Comment by viranch** [comment]
Two ways to fix this:

1. Downgrade `azure-identity` back to 1.5.0 (to address #17) and revert #19 and restrict to `azure-mgmt-trafficmanager==0.51.0` in setup.py

2. Take the leap and merge #21 (bump version in setup.py as well before merging)

I ran quick tests with `azure-mgmt-trafficmanager==1.0.0` in #21 and things seem fine. Will fix as and when we run into bugs, if any.

**Comment by viranch** [comment]
Going ahead with the 2nd way above. The other one is too cumbersome and probably not worth the effort.

> setup.py is allowing the latest version

Need to fix this in a separate PR for good though, major version releases can break things.

**Comment by viranch** [comment]
Fixed by #21
